### ISSUE TITLE
Add commands to pallete for creating a note in a specific folder

### ISSUE THREAD

**Issue by MikhailNatalenko** [SilentVoid13/Templater#1249]
For example I have 3 type of notes: Projects, Resources and Areas (yes it's PARA). I use templates for them. So I have to open the vault, find the right one folder, click somewhere inside and then Add Note if i want to create a note.

I would like to have a list of settings that creates distinct commands for creating a file of specific type.

I can't find any other solution. Sorry maybe there is some plugin, but I have not found it yet.

**Comment by Zachatoo** [comment]
It's easy enough to move the newly created file to a specific folder with `tp.file.move()`, I'd recommend that as the solution.

**Comment by MikhailNatalenko** [comment]
This solution works fine if you have some default folder for all notes. But if you place new notes in the same folder as a current file you will have a strange behaviour, if you try to create `TemplateB-file` in the `TemplateA-folder`. 

Also i think that it is a handy option: if you have some template connected to some folder? plugin stores this setting. It would be great to have a checkpoint "Create file in the specified folder"

Also if there are several folders for one template you can show all options in command palette

**Comment by Zachatoo** [comment]
Maybe I'm misunderstanding what you need, but I still think `tp.file.move()` is a great solution for this. For example, I have a "source" template that looks like this:

```js
<%*
let title = tp.file.title;
if (tp.file.title.startsWith("Untitled")) {
  title = await tp.system.prompt("Title");
}
// Move to sources folder
await tp.file.move(`40 Sources/${title}`);
-%>
<% "---" %>
author:
source:
links:
date:
<% "---" %>

# <% title %>

<% tp.file.cursor() %>
```

And I also have a "term" template that looks like this:
```js
<%*
let title = tp.file.title;
if (tp.file.title.startsWith("Untitled")) {
  title = await tp.system.prompt("Title");
}
// Move to terms folder
await tp.file.move(`30 Terms/${title}`);
-%>
<% "---" %>
aliases:
  - <%* tR += title.toLowerCase() %>
<% "---" %>

# <% title %>

<% tp.file.cursor() %>
```

When I use the "source" template, the note created from the template will be placed in my sources folder. When I use the "term" template, the note created from the template will be placed in my terms folder.

I combine this with the [folder templates](https://silentvoid13.github.io/Templater/settings.html#folder-templates) feature to associate templates to a folder, so if I ever create a note in a folder it will always use a specific template. Though in that case it wouldn't be moved, since it's already in the folder it's supposed to be in.

**Comment by MikhailNatalenko** [comment]
Thank you for the detailed answer, i will try this out in my scenarios and come later)

**Comment by MikhailNatalenko** [comment]
let explain the problem I face (i have double prompts instead of one): 

I have a folder called "Projects" and a folder called "Resources".  Both has it's own templates: `project` and `resource`

- Projects (I'm here
- Resources

Usually I open some `project`, and _then_ create some new `resources` (while i'm still in the Projects folder)


- Projects <- we are here. trying to create resources file
- Resources

By the way I have _folder templates_  as you said. if I run the  command to create a file firstly it applies template 
`project` (prompt for projects pop's up)? run it's sript

- Projects <- it is a `project`, because we are in the projects dir give a prompt to name it
- Resources

and then applies a proper template: `resource` (another prompt pops up)  


- Projects <- well it's time to me the new file to resources dir!
- Resources

result: we have `project` file in the Resources dir

 






If you have setting

**Comment by Zachatoo** [comment]
Sounds like for every new note you create, you want to be prompted for where to put the file and to run a specific template depending on where you want to put it?

That's very similar to my setup that I've documented [here](https://zachyoung.dev/posts/folder-templates-with-quick-switcher). In short, you'll want to add a folder template to the root of your vault that will prompt you for which folder template you want to execute.

**Comment by holroy** [comment]
> I combine this with the [folder templates](https://silentvoid13.github.io/Templater/settings.html#folder-templates) feature to associate templates to a folder, so if I ever create a note in a folder it will always use a specific template. 

So given a file created in the source folder has the source template applied, how do you create a term file when you're in the source folder? 

If you use a create note not linked to a format template, using `tp.file.move()` is applied too late to make the folder template kick in, since the default template has already been applied. And if you create it in the same folder, the source template is applied, not the wanted term template

This is the kind of case, that ask for an opportunity to create a file in a given folder so that the correct folder template is applied. 

---

In other words, when looking at a source file how do you create a note to end up in the term folder, and use the term template? 

Using the mouse you can right-click on the term folder and choose create new note, but how do you do this using the keyboard? 

The two workarounds I know is the right-click method described above, or by first typing a fully qualified link,  and then clicking on that. It would be nice to have a create in folder command (with fuzzy  auto complete) , which would respect the folder templates.

**Comment by Zachatoo** [comment]
@holroy See the message above yours, I've linked to how I do this. Whenever any new note is created, Templater asks me where to put it, and applies the appropriate template.

**Comment by MichalBryxi** [comment]
holroy ***@***.***> schrieb am Do., 22. Feb. 2024, 01:29:

**Comment by MikhailNatalenko** [comment]
@Zachatoo Thank you very much for the reply. It looks pretty interesting. My current approach is quite similar (but raw). I just create a new note in default directory and then manually apply a template (i set a shortcut). There is a `tp.file.move` line in the template. Maybe i will try your approach too.

**Comment by holroy** [comment]
> @holroy See the message above yours, I've linked to how I do this. Whenever any new note is created, Templater asks me where to put it, and applies the appropriate template.

I'm sorry for only skimming your reply above. I briefly read it, and saw it using quick switcher, and I just commented on an auto complete feature for the quick switcher, and furthermore I'm contemplating on how to get notes not only in the main folder where a folder template would suffice, but I want it auto-created within the folder structure. Like having `journal` for the folder template, but `journal/2024/02/2024-02-22.md` for the actual note, or `ForumStuff` for the folder template, but `ForumStuff/f77/f77733 Some title.md` for a forum answer.

So your answer above addresses some of this, and I'm sorry for not reading it more thoroughly, and then again I need to address some other issues as well, like coping with #1327 until it gets fixed.

Lastly, on a totally unrelated note, I loved your note on the [Quick capture for Android](https://zachyoung.dev/posts/obsidian-quick-capture-for-android/), @Zachatoo . So keep up the good work, and I'll go quietly out into the wind and play with some kites.
